File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
 void dfs(struct TreeNode* root,int depth, int* arr,int* returnSize ){
    if(root==NULL){
        return;
    }

if(depth==*returnSize ){
    arr[* returnSize ]=root->val;
     (*returnSize)++;
}
dfs(root->right,depth+1,arr, returnSize );
dfs(root->left,depth+1,arr, returnSize );

 }
int* rightSideView(struct TreeNode* root, int* returnSize) {
    int *arr=malloc(100*sizeof(int));
    if(root==NULL){
        * returnSize=0;
        return NULL;

    }
    * returnSize=0;
    dfs(root, 0, arr,returnSize);
    return arr;
  

    
}

File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        arr=[]
        depth=0
        #size=0
        def dfs(root,arr,depth):
            if root is None:
                return
            if depth ==len(arr):
                arr.append(root.val)
                #s#ize=size+1

            dfs(root.right,arr,depth+1)
            dfs(root.left,arr,depth+1)
            return arr
        dfs(root,arr,depth)
        return arr
        